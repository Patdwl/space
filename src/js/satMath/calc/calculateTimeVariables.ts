import { MILLISECONDS_PER_DAY, MINUTES_PER_DAY } from '@app/js/lib/constants';
import { SatelliteRecord, Sgp4 } from 'ootk';
import { jday } from '../../timeManager/transforms';

export const calculateTimeVariables = (now: Date, satrec?: SatelliteRecord): { gmst: number; m: number; j: number } => {
  const j =
    jday(now.getUTCFullYear(), now.getUTCMonth() + 1, now.getUTCDate(), now.getUTCHours(), now.getUTCMinutes(), now.getUTCSeconds()) +
    now.getUTCMilliseconds() * MILLISECONDS_PER_DAY;
  const gmst = Sgp4.gstime(j);

  const m = satrec ? (j - satrec.jdsatepoch) * MINUTES_PER_DAY : null;

  return { gmst, m, j };
};
