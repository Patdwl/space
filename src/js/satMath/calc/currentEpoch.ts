import { stringPad } from '@app/js/lib/helpers';
import { keepTrackApi } from '../../api/keepTrackApi';

export const currentEpoch = (currentDate: Date): [string, string] => {
  const { timeManager } = keepTrackApi.programs;

  const currentDateObj = new Date(currentDate);
  let epochYear = currentDateObj.getUTCFullYear().toString().substr(2, 2);
  let epochDay = timeManager.getDayOfYear(currentDateObj);
  let timeOfDay = (currentDateObj.getUTCHours() * 60 + currentDateObj.getUTCMinutes()) / 1440;
  const epochDayStr = stringPad.pad0((epochDay + timeOfDay).toFixed(8), 12);
  return [epochYear, epochDayStr];
};
