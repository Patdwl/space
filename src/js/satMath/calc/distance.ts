import { EciPos } from '@app/js/api/keepTrackTypes';

export const distance = (obj1: EciPos, obj2: EciPos): number => {
  const distX = Math.pow(obj1.x - obj2.x, 2);
  const distY = Math.pow(obj1.y - obj2.y, 2);
  const distZ = Math.pow(obj1.z - obj2.z, 2);
  const distAbs = Math.sqrt(distX + distY + distZ);
  return distAbs;
};
