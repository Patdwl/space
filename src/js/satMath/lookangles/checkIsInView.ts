import { SensorObject } from '../../api/keepTrackTypes';

export const checkIsInView = (sensor: SensorObject, rae: { rng: number; az: number; el: number }): boolean => {
  const { az, el, rng } = rae;

  if (sensor.obsminaz > sensor.obsmaxaz) {
    if (
      ((az >= sensor.obsminaz || az <= sensor.obsmaxaz) && el >= sensor.obsminel && el <= sensor.obsmaxel && rng <= sensor.obsmaxrange && rng >= sensor.obsminrange) ||
      ((az >= sensor.obsminaz2 || az <= sensor.obsmaxaz2) && el >= sensor.obsminel2 && el <= sensor.obsmaxel2 && rng <= sensor.obsmaxrange2 && rng >= sensor.obsminrange2)
    ) {
      return true;
    } else {
      return false;
    }
  } else {
    if (
      (az >= sensor.obsminaz && az <= sensor.obsmaxaz && el >= sensor.obsminel && el <= sensor.obsmaxel && rng <= sensor.obsmaxrange && rng >= sensor.obsminrange) ||
      (az >= sensor.obsminaz2 && az <= sensor.obsmaxaz2 && el >= sensor.obsminel2 && el <= sensor.obsmaxel2 && rng <= sensor.obsmaxrange2 && rng >= sensor.obsminrange2)
    ) {
      return true;
    } else {
      return false;
    }
  }
};
