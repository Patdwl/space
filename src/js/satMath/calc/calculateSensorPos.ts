import { DEG2RAD, PLANETARIUM_DIST, RADIUS_OF_EARTH } from '@app/js/lib/constants';
import { keepTrackApi } from '../../api/keepTrackApi';
import { SensorObject } from '../../api/keepTrackTypes';
import { calculateTimeVariables } from './calculateTimeVariables';
import { verifySensors } from './verifySensors';

export const calculateSensorPos = (sensors?: SensorObject[]): { x: number; y: number; z: number; lat: number; lon: number; gmst: number } => {
  const { timeManager, sensorManager } = keepTrackApi.programs;
  sensors = verifySensors(sensors, sensorManager);
  const sensor = sensors[0];

  const now = timeManager.simulationTimeObj;
  const { gmst } = calculateTimeVariables(now);

  const cosLat = Math.cos(sensor.lat * DEG2RAD);
  const sinLat = Math.sin(sensor.lat * DEG2RAD);
  const cosLon = Math.cos(sensor.lon * DEG2RAD + gmst);
  const sinLon = Math.sin(sensor.lon * DEG2RAD + gmst);

  return {
    x: (RADIUS_OF_EARTH + PLANETARIUM_DIST) * cosLat * cosLon,
    y: (RADIUS_OF_EARTH + PLANETARIUM_DIST) * cosLat * sinLon,
    z: (RADIUS_OF_EARTH + PLANETARIUM_DIST) * sinLat,
    gmst: gmst,
    lat: sensor.lat,
    lon: sensor.lon,
  };
};
