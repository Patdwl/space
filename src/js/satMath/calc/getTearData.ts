import { SatelliteRecord } from 'ootk';
import { SensorObject, TearrData } from '../../api/keepTrackTypes';
import { dateFormat } from '../../lib/external/dateFormat.js';
import { checkIsInView } from '../lookangles/checkIsInView';
import { getRae } from './getRae';

export const getTearData = (now: Date, satrec: SatelliteRecord, sensors: SensorObject[], isRiseSetLookangles: boolean): TearrData => {
  // TODO: Instead of doing the first sensor this should return an array of TEARRs for all sensors.
  const sensor = sensors[0];

  let aer = getRae(now, satrec, sensor);
  const isInFOV = checkIsInView(sensor, aer);

  if (isInFOV) {
    if (isRiseSetLookangles) {
      // Previous Pass to Calculate first line of coverage
      const now1 = new Date();
      now1.setTime(Number(now) - 1000);
      let aer1 = getRae(now1, satrec, sensor);
      let isInFOV1 = checkIsInView(sensor, aer1);

      // Is in FOV and Wasn't Last Time so First Line of Coverage
      if (!isInFOV1) {
        return {
          time: dateFormat(now, 'isoDateTime', true),
          rng: aer.rng,
          az: aer.az,
          el: aer.el,
          name: sensor.shortName,
        };
      } else {
        // Next Pass to Calculate Last line of coverage
        now1.setTime(Number(now) + 1000);
        aer1 = getRae(now1, satrec, sensor);
        isInFOV1 = checkIsInView(sensor, aer1);

        // Is in FOV and Wont Be Next Time so Last Line of Coverage
        if (!isInFOV1) {
          return {
            time: dateFormat(now, 'isoDateTime', true),
            rng: aer.rng,
            az: aer.az,
            el: aer.el,
            name: sensor.shortName,
          };
        }
      }
      return {
        time: '',
        rng: -1,
        az: -1,
        el: -1,
        name: sensor.shortName,
      };
    }
    return {
      time: dateFormat(now, 'isoDateTime', true),
      rng: aer.rng,
      az: aer.az,
      el: aer.el,
      name: sensor.shortName,
    };
  }
  return {
    time: '',
    rng: -1,
    az: -1,
    el: -1,
    name: sensor.shortName,
  };
};
