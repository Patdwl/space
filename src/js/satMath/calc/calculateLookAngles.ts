import { DEG2RAD } from '@app/js/lib/constants';
import { calcSatrec } from '@app/js/satSet/catalogSupport/calcSatrec';
import { keepTrackApi } from '../../api/keepTrackApi';
import { SatObject, SensorObject, TearrData } from '../../api/keepTrackTypes';
import { satellite } from '../satMath';
import { getTearData } from './getTearData';

export const calculateLookAngles = (sat: SatObject, sensors: SensorObject[]): TearrData[] => {
  const { sensorManager, timeManager } = keepTrackApi.programs;

  (function _inputValidation() {
    // Check if there is a sensor
    if (typeof sensors == 'undefined') {
      // Try using the current sensor if there is one
      if (sensorManager.checkSensorSelected()) {
        sensors = sensorManager.currentSensor;
      } else {
        console.debug('getlookangles2 requires a sensor!');
        return;
      }
      // Simple Error Checking
    } else {
      if (typeof sensors[0].obsminaz == 'undefined') {
        console.debug('sensors[0] format incorrect');
        return;
      }
      sensors[0].observerGd = {
        // Array to calculate look angles in propagate()
        lat: sensors[0].lat * DEG2RAD,
        lon: sensors[0].lon * DEG2RAD,
        alt: sensors[0].alt,
      };
    }

    if (typeof sat == 'undefined') {
      console.debug('sat parameter required!');
    } else {
      if (typeof sat.TLE1 == 'undefined' || typeof sat.TLE2 == 'undefined') {
        console.debug('sat parameter invalid format!');
      }
    }

    if (typeof satellite.isRiseSetLookangles == 'undefined') {
      satellite.isRiseSetLookangles = false;
    }
  })();

  // TOOD: Instead of doing the first sensor this should return an array of TEARRs for all sensors.
  const sensor = sensors[0];

  const simulationTime = timeManager.simulationTimeObj;
  let offset = 0;
  var satrec = calcSatrec(sat);
  var lookanglesTable = []; // Iniially no rows to the table
  var tempLookanglesInterval;

  if (satellite.isRiseSetLookangles) {
    tempLookanglesInterval = satellite.lookanglesInterval;
    satellite.lookanglesInterval = 1;
  }

  for (var i = 0; i < satellite.lookanglesLength * 24 * 60 * 60; i += satellite.lookanglesInterval) {
    // satellite.lookanglesInterval in seconds
    offset = i * 1000; // Offset in seconds (msec * 1000)
    const now = timeManager.getOffsetTimeObj(offset, simulationTime);
    if (lookanglesTable.length <= 5000) {
      // Maximum of 1500 lines in the look angles table
      const _lookanglesRow = getTearData(now, satrec, [sensor], satellite.isRiseSetLookangles);
      if (_lookanglesRow.time !== '') {
        lookanglesTable.push(_lookanglesRow); // Update the table with looks for this 5 second chunk and then increase table counter by 1
      }
    }
  }

  if (satellite.isRiseSetLookangles) {
    satellite.lookanglesInterval = tempLookanglesInterval;
  }
  return lookanglesTable;
};
