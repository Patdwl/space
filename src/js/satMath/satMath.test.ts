// TODO: Jest snapshots are currently dependent on local computer time. They
// should be updated to use the same time on all computers.

import { defaultSat, defaultSensor, keepTrackApiStubs } from '../api/apiMocks';
import { keepTrackApi } from '../api/keepTrackApi';
import { KeepTrackPrograms, ZoomValue } from '../api/keepTrackTypes';
import { SpaceObjectType } from '../api/SpaceObjectType';
import * as getTearData from './calc/getTearData';
import * as checkIsInView from './lookangles/checkIsInView';
import * as satMath from './satMath';
import { satellite } from './satMath';

keepTrackApi.programs = <KeepTrackPrograms>(<unknown>{ ...keepTrackApi.programs, ...keepTrackApiStubs.programs });
export const dateNow = new Date(2022, 0, 1);
dateNow.setUTCHours(0, 0, 0, 0);

describe('satMath.getTearData', () => {
  const realSgp4 = satellite.sgp4;
  beforeAll(() => {
    satellite.sgp4 = () =>
      <any>{
        position: {
          x: 6000,
          y: 6000,
          z: 6000,
        },
      };
  });
  it('should handle isInFOV', () => {
    jest.spyOn(checkIsInView, 'checkIsInView').mockImplementationOnce(() => true);
    let result: any = getTearData.getTearData(dateNow, satellite.twoline2satrec(defaultSat.TLE1, defaultSat.TLE2), [defaultSensor], false);
    expect(result).toMatchSnapshot();
  });
  it('should handle isInFOV and isRiseSetLookangles', () => {
    jest.spyOn(checkIsInView, 'checkIsInView').mockImplementationOnce(() => true);
    let result: any = getTearData.getTearData(dateNow, satellite.twoline2satrec(defaultSat.TLE1, defaultSat.TLE2), [defaultSensor], true);
    expect(result).toMatchSnapshot();
  });
  afterAll(() => {
    satellite.sgp4 = realSgp4;
  });

  it('should handle isInFOV and isRiseSetLookangles not previously in FOV but will be', () => {
    jest
      .spyOn(checkIsInView, 'checkIsInView')
      .mockImplementationOnce(() => true)
      .mockImplementationOnce(() => false);
    let result: any = getTearData.getTearData(dateNow, satellite.twoline2satrec(defaultSat.TLE1, defaultSat.TLE2), [defaultSensor], true);
    expect(result).toMatchSnapshot();
    jest
      .spyOn(checkIsInView, 'checkIsInView')
      .mockImplementationOnce(() => true)
      .mockImplementationOnce(() => true);
    result = getTearData.getTearData(dateNow, satellite.twoline2satrec(defaultSat.TLE1, defaultSat.TLE2), [defaultSensor], true);
    expect(result).toMatchSnapshot();
  });
  afterAll(() => {
    satellite.sgp4 = realSgp4;
  });
});

// @ponicode
describe('satMath.setobs', () => {
  test('0', () => {
    let result: any = satMath.setobs([
      {
        alt: 0,
        beamwidth: 15,
        changeObjectInterval: -100,
        country: 'US',
        lat: 550,
        linkAehf: false,
        linkWgs: false,
        lon: 50,
        name: 'Pierre Edouard',
        observerGd: { lat: 520, lon: 550, alt: -100 },
        obsmaxaz: 1,
        obsmaxel: -29.45,
        obsmaxrange: 380,
        obsminaz: 1,
        obsminel: 0.0,
        obsminrange: 1,
        shortName: '333173976',
        staticNum: 1,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.BALLISTIC_MISSILE,
        url: 'http://base.com',
        volume: false,
        zoom: 'geo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = satMath.setobs([
      {
        alt: -100,
        beamwidth: 40,
        changeObjectInterval: 0,
        country: 'China',
        lat: 410,
        linkAehf: true,
        linkWgs: true,
        lon: 380,
        name: 'Edmond',
        observerGd: { lat: 400, lon: 320, alt: 100 },
        obsmaxaz: 100,
        obsmaxel: -29.45,
        obsmaxrange: 50,
        obsminaz: -100,
        obsminel: 10.23,
        obsminrange: -100,
        shortName: 'test',
        staticNum: 1000,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.PAYLOAD,
        url: 'http://base.com',
        volume: true,
        zoom: 'leo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = satMath.setobs([
      {
        alt: -100,
        beamwidth: 30,
        changeObjectInterval: 1,
        country: 'France',
        lat: 30,
        linkAehf: true,
        linkWgs: false,
        lon: 30,
        name: 'Jean-Philippe',
        observerGd: { lat: 550, lon: 1, alt: 0 },
        obsmaxaz: 1,
        obsmaxel: -0.5,
        obsmaxrange: 380,
        obsminaz: 0,
        obsminel: 1.0,
        obsminrange: -5.48,
        shortName: 'test',
        staticNum: 10,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.PAYLOAD,
        url: 'Www.GooGle.com',
        volume: false,
        zoom: 'leo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = satMath.setobs([
      {
        alt: 0,
        beamwidth: 16,
        changeObjectInterval: -100,
        country: 'China',
        lat: 400,
        linkAehf: true,
        linkWgs: false,
        lon: 30,
        name: 'Jean-Philippe',
        observerGd: { lat: 90, lon: 70, alt: -5.48 },
        obsmaxaz: 0,
        obsmaxel: 0.0,
        obsmaxrange: 350,
        obsminaz: -5.48,
        obsminel: 0.0,
        obsminrange: 1,
        shortName: 'test',
        staticNum: 10,
        sun: 'v4.0.0-rc.4',
        type: SpaceObjectType.PAYLOAD,
        url: 'http://example.com/showcalendar.html?token=CKF50YzIHxCTKMAg',
        volume: false,
        zoom: 'leo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = satMath.setobs([
      {
        alt: 0,
        beamwidth: 5,
        changeObjectInterval: -100,
        country: 'FR',
        lat: 520,
        linkAehf: false,
        linkWgs: true,
        lon: 320,
        name: 'Edmond',
        observerGd: { lat: 50, lon: 50, alt: -5.48 },
        obsmaxaz: 1,
        obsmaxel: 10.23,
        obsmaxrange: 70,
        obsminaz: 100,
        obsminel: 0.5,
        obsminrange: -100,
        shortName: 'test',
        staticNum: 10,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.PAYLOAD,
        url: 'https://accounts.google.com/o/oauth2/revoke?token=%s',
        volume: true,
        zoom: 'leo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = satMath.setobs([
      {
        alt: NaN,
        beamwidth: NaN,
        changeObjectInterval: NaN,
        country: '',
        lat: NaN,
        linkAehf: false,
        linkWgs: true,
        lon: NaN,
        name: '',
        observerGd: { lat: NaN, lon: NaN, alt: NaN },
        obsmaxaz: NaN,
        obsmaxel: NaN,
        obsmaxrange: NaN,
        obsminaz: NaN,
        obsminel: NaN,
        obsminrange: NaN,
        shortName: '',
        staticNum: NaN,
        sun: '',
        type: SpaceObjectType.PAYLOAD,
        url: '',
        volume: false,
        zoom: <ZoomValue>'',
      },
    ]);
    expect(result).toMatchSnapshot();
  });

  test('6', () => {
    let result: any = satMath.setobs([
      {
        alt: 0,
        beamwidth: 5,
        changeObjectInterval: -100,
        country: 'FR',
        lat: 520,
        linkAehf: false,
        linkWgs: true,
        lon: 320,
        name: 'Edmond',
        observerGd: { lat: 50, lon: 50, alt: -5.48 },
        obsmaxaz: 1,
        obsmaxel: 10.23,
        obsmaxrange: 70,
        obsminaz: 100,
        obsminel: 0.5,
        obsminrange: -100,
        shortName: 'test',
        staticNum: 10,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.PAYLOAD,
        url: 'https://accounts.google.com/o/oauth2/revoke?token=%s',
        volume: true,
        zoom: 'leo',
      },
      {
        alt: 0,
        beamwidth: 5,
        changeObjectInterval: -100,
        country: 'FR',
        lat: 520,
        linkAehf: false,
        linkWgs: true,
        lon: 320,
        name: 'Edmond',
        observerGd: { lat: 50, lon: 50, alt: -5.48 },
        obsmaxaz: 1,
        obsmaxel: 10.23,
        obsmaxrange: 70,
        obsminaz: 100,
        obsminel: 0.5,
        obsminrange: -100,
        shortName: 'test',
        staticNum: 10,
        sun: '4.0.0-beta1\t',
        type: SpaceObjectType.PAYLOAD,
        url: 'https://accounts.google.com/o/oauth2/revoke?token=%s',
        volume: true,
        zoom: 'leo',
      },
    ]);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satMath.setTEARR', () => {
  test('0', () => {
    let result: any = satMath.setTEARR({
      alt: 0,
      lat: 0,
      lon: 0,
      rng: 0,
      az: 0,
      el: 0,
      name: '',
    });
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satMath.getTEARR', () => {
  test('0', () => {
    satMath.getTEARR(defaultSat);
  });
});
