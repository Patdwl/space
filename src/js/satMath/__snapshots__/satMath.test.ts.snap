// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`satMath.calculateDops 0 1`] = `
Object {
  "gdop": "2.21",
  "hdop": "1.06",
  "pdop": "2.00",
  "tdop": "0.92",
  "vdop": "1.70",
}
`;

exports[`satMath.calculateDops 1 1`] = `
Object {
  "gdop": 50,
  "hdop": 50,
  "pdop": 50,
  "tdop": 50,
  "vdop": 50,
}
`;

exports[`satMath.calculateVisMag 0 1`] = `NaN`;

exports[`satMath.calculateVisMag 1 1`] = `NaN`;

exports[`satMath.calculateVisMag 2 1`] = `NaN`;

exports[`satMath.calculateVisMag 3 1`] = `NaN`;

exports[`satMath.calculateVisMag 4 1`] = `NaN`;

exports[`satMath.createTle 0 1`] = `
Object {
  "TLE1": "1 25544U 1998-01A 20123  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544 010.0100 010.5000 0.001 001.0000 010.5000 15.15000000    10",
}
`;

exports[`satMath.currentEpoch 0 1`] = `
Array [
  "N",
  "000000000NaN",
]
`;

exports[`satMath.currentEpoch 4 1`] = `
Array [
  "N",
  "000000000NaN",
]
`;

exports[`satMath.distance 0 1`] = `"<br />Range: 0 km"`;

exports[`satMath.distance 1 1`] = `"<br />Range: 0 km"`;

exports[`satMath.findBestPasses 0 1`] = `undefined`;

exports[`satMath.getDops 0 1`] = `
Object {
  "gdop": 50,
  "hdop": 50,
  "pdop": 50,
  "tdop": 50,
  "vdop": 50,
}
`;

exports[`satMath.getOrbitByLatLon 0 1`] = `
Array [
  "Error",
  "Failed to find a solution for Argument of Perigee",
]
`;

exports[`satMath.getSunDirection 0 1`] = `
Array [
  148724320.43455538,
  6189145.714033483,
  14909516.932049064,
]
`;

exports[`satMath.getSunDirection 1 1`] = `
Array [
  1306788.4922269185,
  -120881774.59456515,
  -88121572.7460775,
]
`;

exports[`satMath.getSunTimes 0 1`] = `undefined`;

exports[`satMath.getSunTimes 1 1`] = `undefined`;

exports[`satMath.getTearData should handle isInFOV 1`] = `
Object {
  "az": 113.80780075611027,
  "el": 57.48814771937549,
  "name": "COD",
  "rng": 4439.658324055251,
  "time": "2022-01-01 00:00:00",
}
`;

exports[`satMath.getTearData should handle isInFOV and isRiseSetLookangles 1`] = `
Object {
  "az": -1,
  "el": -1,
  "name": "COD",
  "rng": -1,
  "time": "",
}
`;

exports[`satMath.getTearData should handle isInFOV and isRiseSetLookangles not previously in FOV but will be 1`] = `
Object {
  "az": 113.80780075611027,
  "el": 57.48814771937549,
  "name": "COD",
  "rng": 4439.658324055251,
  "time": "2022-01-01 00:00:00",
}
`;

exports[`satMath.getTearData should handle isInFOV and isRiseSetLookangles not previously in FOV but will be 2`] = `
Object {
  "az": -1,
  "el": -1,
  "name": "COD",
  "rng": -1,
  "time": "",
}
`;

exports[`satMath.getlookangles 0 1`] = `Array []`;

exports[`satMath.getlookangles 1 1`] = `Array []`;

exports[`satMath.getlookangles 2 1`] = `
Array [
  Object {
    "az": 50,
    "el": 30,
    "name": "COD",
    "rng": 1000,
    "time": "2022-01-01T00:00:00.000Z",
  },
]
`;

exports[`satMath.getlookanglesMultiSite 0 1`] = `undefined`;

exports[`satMath.lookAngles2Ecf 0 1`] = `
Object {
  "x": 3399.2987560938654,
  "y": 6013.8733407148075,
  "z": -2379.043164190548,
}
`;

exports[`satMath.lookAngles2Ecf 1 1`] = `
Object {
  "x": 3697.567569850421,
  "y": 7959.494204847528,
  "z": 3243.7908022872425,
}
`;

exports[`satMath.populateMultiSiteTable 0 1`] = `undefined`;

exports[`satMath.setTEARR 0 1`] = `undefined`;

exports[`satMath.setobs 0 1`] = `undefined`;

exports[`satMath.setobs 1 1`] = `undefined`;

exports[`satMath.setobs 2 1`] = `undefined`;

exports[`satMath.setobs 3 1`] = `undefined`;

exports[`satMath.setobs 4 1`] = `undefined`;

exports[`satMath.setobs 5 1`] = `undefined`;

exports[`satMath.setobs 6 1`] = `undefined`;

exports[`satMath.updateDopsTable 0 1`] = `undefined`;
